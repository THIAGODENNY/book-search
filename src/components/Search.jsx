import React, { useEffect } from 'react';
import { DebounceInput } from 'react-debounce-input';
import axios from 'axios';
import { useSelector, useDispatch } from 'react-redux';
import Items from './Items';
import '../styles/components/Search.css';
import SelectList from './SelectList';

function Search() {
  const dispatch = useDispatch();

  const {
    data,
    wishList,
    search,
    selectedOption,
  } = useSelector((state) => state);

  const [
    setData,
    setWishlist,
    setSearch,
    setSelectedOption,
  ] = [
    (newData) => dispatch({ type: 'SET_DATA', data: newData }),
    (newWishList) => dispatch({ type: 'SET_WISHLIST', wishList: newWishList }),
    (newSearch) => dispatch({ type: 'SET_SEARCH', search: newSearch }),
    (newSelectedOption) => dispatch({ type: 'SET_SELECTED_OPTION', selectedOption: newSelectedOption }),
  ];

  const fetchData = async (searchParameters) => {
    setData({ items: [] });
    if (searchParameters) {
      const result = await axios(
        `https://www.googleapis.com/books/v1/volumes?q=${searchParameters}`, {
          headers: {
            'Content-Type': 'application/json;charset=utf-8',
          },
        },
      );
      const { items } = result.data;
      if (items) {
        setData({ items });
      }
    }
  };

  const searchHandle = (event) => {
    setSearch(event);
    fetchData(event);
  };

  useEffect(
    () => localStorage.setItem('items', JSON.stringify(wishList.items)),
  );

  const addItemWishlist = (id) => {
    setSelectedOption({ isOpened: true, id });
  };

  const onRequestClose = () => {
    setSelectedOption({ isOpened: false });
  };

  const removeItemWishList = (id) => {
    setWishlist({ items: wishList.items.filter((i) => i.id !== id) });
  };

  const filterData = () => {
    const { items } = wishList;

    if (items.length > 0 && data.items) {
      return data.items.filter((item) => items.filter(
        (wishListItem) => item.id === wishListItem.id,
      )
        .length === 0);
    }
    return data.items;
  };

  const debounceInputClasses = [
    'search',
    'search-found',
  ];

  return (
    <div className="app-container">
      <div className={
        (search || wishList.items.length > 0)
          ? debounceInputClasses[1]
          : debounceInputClasses[0]
      }
      >
        <DebounceInput
          minLength={2}
          onChange={(event) => searchHandle(event.target.value)}
          debounceTimeout={300}
          className="input-search"
        />
      </div>

      <div className="items-container">
        <div className="items">
          {data.items.length > 0 && <h1 className="title">Items found</h1>}
          <Items items={(() => filterData())()} addItemWishlist={addItemWishlist} />
        </div>
        {wishList.items.length > 0
          && (
          <div className="items">
            <div className="items-wishlist">
              <h1 className="title">WishList</h1>
              <Items items={wishList.items} addItemWishlist={removeItemWishList} />
            </div>
          </div>
          )}
      </div>
      <SelectList
        selectedOption={selectedOption}
        onRequestClose={onRequestClose}
      />
    </div>
  );
}
export default Search;
