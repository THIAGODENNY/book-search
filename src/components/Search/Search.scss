.search {
  width: 100%;
  &__search { 
    grid-area: search-grid;
    margin-top: 40vh;
    height: 50px;
    left: 50%;

    width: 40%;
    @media (max-width: 850px) {
      width: 60%;
    }
  
    animation-name: search__moveDown;
    animation-duration: 600ms;

    &--found {
      margin-top: 5vh;
      height: 50px; 
      left: 50%;

      width: 40%;
      @media (max-width: 850px) {
        width: 60%;
      }
    }

    &__input-search {
      border-radius: 50px;
      width: 100%;
      height: 100%;
      font-size: 150%;
      text-indent: 18px;

      &:hover {
        border-color: blue;
      }

      &[type="text"]:focus {
        border-color: blue;
        outline: none;
      }
    }
  }

  &__items {
    display: grid;
    grid-template-columns: 100%;
    width: 100%;

    &__found {
      &__title {
        font-size: x-large;
        text-align: initial;
        margin: 15px;
        margin-top: 10px;
        margin-bottom: 0;
      }
    }

    &__wishlist {
      width: 100%;
      &__header {
        display: flex;
        margin: 0;
        align-items: center;

        &__title {
          font-size: x-large;
          text-align: initial;
          margin: 15px;
          margin-top: 10px;
          margin-bottom: 0;
        }

        &__filter {
          margin: 0;
        }
      }
    }
  }
}

.modal__test__from__scratch {
  background-color: red;
  position: absolute;
  display: flex;
  align-items: center;
  top: 10%;
  left: 10%;
  right: 10%;
  bottom: 10%;
  height: 80%;
  width: 80%;
}

@keyframes search__moveDown {
  0% {
    opacity: 0;
    margin-top: -10vh;
  }

  100% {
    opacity: 1;
    margin-top: 40vh;
  }
}
