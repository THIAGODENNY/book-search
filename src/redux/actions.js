import axios from 'axios';
import store from './store';

const setData = (data) => ({ type: 'SET_DATA', data });
const setWishList = (wishList) => ({ type: 'SET_WISHLIST', wishList });
const setSearch = (search) => ({ type: 'SET_SEARCH', search });
const setSelectedOption = (selectedOption) => ({ type: 'SET_SELECTED_OPTION', selectedOption });
const setFilter = (filter) => ({ type: 'SET_FILTER', filter });
const setAllList = (list) => ({ type: 'SET_ALL_LIST', list });
const setList = (list) => ({ type: 'SET_LIST', list });
const setCreateListIsOpen = (createListIsOpen) => ({ type: 'SET_CREATE_LIST_IS_OPEN', createListIsOpen });
const setResetPage = () => ({ type: 'RESET_PAGE' });
const setIncrementPage = () => ({ type: 'INCREMENT_PAGE' });
const setHasMoreItems = () => ({ type: 'SET_HAS_MORE_ITEMS' });
const resetHasMoreItems = () => ({ type: 'RESET_HAS_MORE_ITEMS' });

export const updateItemsData = (item) => store.dispatch(setData(item));
export const updateWishList = (item) => {
  localStorage.setItem('items', JSON.stringify(item.items));
  store.dispatch(setWishList(item));
};
export const updateSearch = (item) => store.dispatch(setSearch(item));
export const updateSelectedOption = (option) => store.dispatch(setSelectedOption(option));
export const updateFilter = (filter) => store.dispatch(setFilter(filter));
export const updateAllList = (list) => store.dispatch(setAllList(list));
export const updateList = (list) => store.dispatch(setList(list));
export const updateCreateListIsOpen = (createListIsOpen) => store.dispatch(
  setCreateListIsOpen(createListIsOpen),
);
export const resetPage = () => store.dispatch(setResetPage());
export const incrementPage = () => store.dispatch(setIncrementPage());
export const updateHasMoreItems = (state) => {
  if (state) {
    return store.dispatch(setHasMoreItems());
  }
  return store.dispatch(resetHasMoreItems());
};

const fetchData = async () => {
  const { page, data, search } = store.getState();
  if (search) {
    const result = await axios(
      `https://www.googleapis.com/books/v1/volumes?q=${search}&startIndex=${page}`, {
        headers: {
          'Content-Type': 'application/json;charset=utf-8',
        },
      },
    );
    const { items } = result.data;
    if (items) {
      if (page === 0) {
        updateHasMoreItems(true);
        updateItemsData({ items });
      } else {
        const newItems = data.items.concat(items);
        updateItemsData({ items: newItems });
      }
      incrementPage();
    } else {
      updateHasMoreItems(false);
    }
  }
};

export const getMorePages = () => {
  fetchData();
};

export const searchHandle = (event) => {
  updateItemsData({ items: [] });
  resetPage();
  updateSearch(event);
  fetchData();
};

export const handleFilter = (e) => {
  const selectedFilter = e.target.value;
  if (selectedFilter === 'null' || !selectedFilter) {
    updateFilter(null);
  } else {
    updateFilter(selectedFilter);
  }
};

export const addItemWishlist = (id) => (updateSelectedOption({ isOpened: true, id }));
export const onRequestClose = () => (updateSelectedOption({ isOpened: false }));
export const removeItemWishList = (items) => (updateWishList({ items }));

export const handleCreateListClose = () => {
  updateCreateListIsOpen(false);
};

export const handleCreateList = (e) => {
  e.preventDefault();
  updateCreateListIsOpen(true);
};
