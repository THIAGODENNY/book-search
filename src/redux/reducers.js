const initialState = {
  data: {
    items: [],
  },
  page: 0,
  wishList: {
    items: JSON.parse(localStorage.getItem('items') || '[]'),
  },
  search: undefined,
  list: [],
  selectedOption: {
    isOpen: false,
    id: undefined,
  },
  createListIsOpen: false,
  filter: undefined,
  hasMoreItems: true,
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case 'SET_DATA':
      return {
        ...state,
        data: action.data,
      };
    case 'INCREMENT_PAGE':
      return {
        ...state,
        page: state.page + 10,
      };
    case 'RESET_PAGE':
      return {
        ...state,
        page: 0,
      };
    case 'SET_WISHLIST':
      return {
        ...state,
        wishList: action.wishList,
      };
    case 'SET_SEARCH':
      return {
        ...state,
        search: action.search,
      };
    case 'SET_LIST':
      return {
        ...state,
        list: [...state.list, action.list],
      };
    case 'SET_ALL_LIST':
      return {
        ...state,
        list: action.list,
      };
    case 'SET_SELECTED_OPTION':
      return {
        ...state,
        selectedOption: action.selectedOption,
      };
    case 'SET_CREATE_LIST_IS_OPEN':
      return {
        ...state,
        createListIsOpen: action.createListIsOpen,
      };
    case 'SET_FILTER':
      return {
        ...state,
        filter: action.filter,
      };
    case 'SET_HAS_MORE_ITEMS':
      return {
        ...state,
        hasMoreItems: true,
      };
    case 'RESET_HAS_MORE_ITEMS':
      return {
        ...state,
        hasMoreItems: false,
      };
    default:
      return state;
  }
};

export default reducer;
