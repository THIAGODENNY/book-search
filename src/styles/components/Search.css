.search, .search-found {
  grid-area: search-grid;
}

.items-container {
  grid-area: items-container-grid;
}

.app {
  display: grid;
  grid-template-areas: 'search-grid search-grid search-grid'
                       'items-container-grid items-container-grid items-container-grid';

}

.search {
  margin-top: 40vh;

  height: 50px;
  width: 40%;
  left: 50%;

  animation-name: moveDown;
  animation-duration: 600ms;
}

.search-found {
  margin-top: 5vh;

  height: 50px;
  width: 40%;
  left: 50%;
}

.input-search {
  border-radius: 50px;
  width: 100%;
  height: 100%;
  font-size: 150%;
  text-indent: 18px;
}

input:hover {
  border-color: blue;
}

input[type="text"]:focus {
  border-color: blue;
  outline: none;
}

.items-container {
  display: grid;
  grid-template-columns: 50% 50%;
  width: 100%;
}

@keyframes moveDown {
  0% {
    opacity: 0;
    margin-top: -10vh;
  }

  100% {
    opacity: 1;
    margin-top: 40vh;
  }
}
