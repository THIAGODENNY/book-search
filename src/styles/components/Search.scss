.search {
  &__search {
    grid-area: search-grid;
    margin-top: 40vh;

    height: 50px;
    width: 40%;
    left: 50%;
  
    animation-name: search__moveDown;
    animation-duration: 600ms;

    &--found {
      margin-top: 5vh;

      height: 50px;
      width: 40%;
      left: 50%;
    }

    &__input-search {
      border-radius: 50px;
      width: 100%;
      height: 100%;
      font-size: 150%;
      text-indent: 18px;

      &:hover {
        border-color: blue;
      }

      &[type="text"]:focus {
        border-color: blue;
        outline: none;
      }
    }
  }

  &__items {
    display: grid;
    grid-template-columns: 50% 50%;
    width: 100%;

    &__found {
      &__title {
        font-size: x-large;
        text-align: initial;
        margin: 15px;
        margin-top: 10px;
        margin-bottom: 0;
      }
    }

    &__wishlist {
      &__header {
        display: flex;
        margin: 0;
        align-items: center;

        &__title {
          font-size: x-large;
          text-align: initial;
          margin: 15px;
          margin-top: 10px;
          margin-bottom: 0;
        }

        &__filter {
          margin: 0;
        }
      }
    }
  }
}

@keyframes search__moveDown {
  0% {
    opacity: 0;
    margin-top: -10vh;
  }

  100% {
    opacity: 1;
    margin-top: 40vh;
  }
}
