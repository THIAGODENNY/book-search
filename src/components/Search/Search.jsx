import React from 'react';
import PropTypes, { object } from 'prop-types';
import { DebounceInput } from 'react-debounce-input';
import { connect } from 'react-redux';
import InfiniteScroll from 'react-infinite-scroller';
import classNames from 'classnames';
import Items from '../Items';
import './Search.scss';
import SelectList from '../SelectList';
import arraysEqual from '../../../tools/arraysEqual';

import {
  updateFilter,
  updateAllList,
  searchHandle,
  addItemWishlist,
  onRequestClose,
  getMorePages,
} from '../../redux/actions';

class Search extends React.PureComponent {
  componentDidMount() {
    this.update();
  }

  update() {
    const {
      wishList, list, updateFilterDispatch, updateAllListDispatch,
    } = this.props;
    const { items } = wishList;

    localStorage.setItem('list', list.join(','));

    const wishListLists = items.map((item) => item.listName);
    const newAllList = [...new Set(wishListLists)];

    if (
      !arraysEqual(list, newAllList)
    ) {
      updateAllListDispatch(newAllList);
      updateFilterDispatch(null);
    }
  }

  filterData() {
    const {
      wishList,
      data,
      hasMoreItems,
      getMorePagesDispatch,
    } = this.props;

    const { items } = wishList;

    if (items.length > 0 && data.items) {
      const itemsToLoad = data.items.filter((item) => items.filter(
        (wishListItem) => item.id === wishListItem.id,
      )
        .length === 0);
      if (items.length > 0 && itemsToLoad.length < 10 && hasMoreItems) {
        getMorePagesDispatch();
      }
      return itemsToLoad;
    }
    return data.items;
  }

  render() {
    const {
      selectedOption,
      search,
      data,
      hasMoreItems,
      getMorePagesDispatch,
      searchHandleDispatch,
      addItemWishlistDispatch,
      onRequestCloseDispatch,
    } = this.props;

    return (
      <div data-test="search" className="search">
        <div className={
          classNames({ search__search: !search, 'search__search--found': search })
        }
        >
          <DebounceInput
            minLength={2}
            onChange={(event) => searchHandleDispatch(event.target.value)}
            debounceTimeout={1000}
            data-search="search__search__input-search"
            className="search__search__input-search"
            value={search}
          />
        </div>

        <div data-test="search__items" className="search__items">
          <div data-test="search__items__found" className="search__items__found">
            {search
              && (
                <div>
                  {data.items.length > 0 && <h1 data-test="search__items__found__title" className="search__items__found__title">Items found</h1>}
                  <InfiniteScroll
                    data-test="search__items__found__infinite__scroll"
                    className="search__items__found__infinite__scroll"
                    initialLoad={false}
                    pageStart={0}
                    loadMore={() => getMorePagesDispatch()}
                    hasMore={hasMoreItems}
                    loader={data.items.length > 0 && <div className="loader" key={0}>Loading ...</div>}
                  >
                    <Items data-test="search__items__component" items={(() => this.filterData())()} addItemWishlist={addItemWishlistDispatch} />
                  </InfiniteScroll>
                </div>
              )}
          </div>
        </div>
        <SelectList
          data-test="item__select-list"
          className="item__select-list"
          selectedOption={selectedOption}
          onRequestClose={onRequestCloseDispatch}
        />
      </div>
    );
  }
}

Search.propTypes = {
  wishList: PropTypes.shape({
    items: PropTypes.arrayOf(object).isRequired,
  }).isRequired,
  data: PropTypes.shape({
    items: PropTypes.arrayOf(PropTypes.object),
  }).isRequired,
  list: PropTypes.arrayOf(PropTypes.string).isRequired,
  hasMoreItems: PropTypes.bool.isRequired,
  search: PropTypes.string.isRequired,
  selectedOption: PropTypes.shape({
    isOpen: PropTypes.bool.isRequired,
    id: PropTypes.string.isRequired,
  }).isRequired,
  getMorePagesDispatch: PropTypes.func.isRequired,
  searchHandleDispatch: PropTypes.func.isRequired,
  updateFilterDispatch: PropTypes.func.isRequired,
  updateAllListDispatch: PropTypes.func.isRequired,
  addItemWishlistDispatch: PropTypes.func.isRequired,
  onRequestCloseDispatch: PropTypes.func.isRequired,
};

Search.defaults = {
  selectedOption: {
    isOpen: false,
    id: '',
  },
};

const mapStateToProps = (state) => ({
  data: state.data,
  wishList: state.wishList,
  search: state.search,
  selectedOption: state.selectedOption,
  list: state.list,
  hasMoreItems: state.hasMoreItems,
});

const mapDispatchToProps = (dispatch) => ({
  getMorePagesDispatch: () => dispatch(getMorePages),
  searchHandleDispatch: (event) => dispatch(searchHandle(event)),
  updateFilterDispatch: () => dispatch(updateFilter),
  updateAllListDispatch: (event) => dispatch(updateAllList(event)),
  addItemWishlistDispatch: (event) => dispatch(addItemWishlist(event)),
  onRequestCloseDispatch: () => dispatch(onRequestClose),
});

export default connect(mapStateToProps, mapDispatchToProps)(Search);
