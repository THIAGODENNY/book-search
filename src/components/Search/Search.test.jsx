import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import Search from './Search';
import { getMorePages } from '../../redux/actions';
import { storeFactory } from '../../tests/storeFactory';

const initialState = (props) => ({
  data: {
    items: [],
  },
  page: 0,
  wishList: {
    items: JSON.parse(localStorage.getItem('items') || '[]'),
  },
  search: '',
  list: [],
  selectedOption: {
    isOpen: false,
    id: '',
  },
  createListIsOpen: false,
  filter: undefined,
  hasMoreItems: true,
  ...props,
});

const mockStore = configureStore([]);
const setup = (props) => {
  const store = mockStore(initialState(props));

  return shallow(
    <Search
      store={store}
    />,
  )
    .dive()
    .dive();
};

describe('Search should render correctly ', () => {
  it('render search component', () => {
    const component = setup();
    const itemSearch = component.find('[data-test="search"]');
    expect(itemSearch.length).toBe(1);
  });

  it('render search__search component', () => {
    const component = setup();
    const itemSearch = component.find('[className="search__search"]');
    expect(itemSearch.length).toBe(1);
  });
  it('render search__search--found component when search isn`t empty', () => {
    const component = setup({ search: 'fooBar' });
    const itemSearch = component.find('[className="search__search--found"]');
    expect(itemSearch.length).toBe(1);
  });
  it('render search__search__input-search component', () => {
    const component = setup();
    const itemSearch = component.find('[className="search__search__input-search"]');
    expect(itemSearch.length).toBe(1);
  });

  it('render search__items component', () => {
    const component = setup();
    const itemSearch = component.find('[data-test="search__items"]');
    expect(itemSearch.length).toBe(1);
  });

  it('render search__items__found component', () => {
    const component = setup();
    const itemSearch = component.find('[data-test="search__items__found"]');
    expect(itemSearch.length).toBe(1);
  });

  it('not render search__items__found__title when search is empty', () => {
    const component = setup({ search: '', data: { items: [{ id: 'fooBar' }] } });
    const itemSearch = component.find('[data-test="search__items__found__title"]');
    expect(itemSearch.length).toBe(0);
  });

  it('not render search__items__found__title when data.items is empty', () => {
    const component = setup({ search: 'fooBar', data: { items: [] } });
    const itemSearch = component.find('[data-test="search__items__found__title"]');
    expect(itemSearch.length).toBe(0);
  });

  it('render search__items__found__title when search and data.items isnt`t empty', () => {
    const component = setup({ search: 'fooBar', data: { items: [{ id: 'fooBar' }] } });
    const itemSearch = component.find('[data-test="search__items__found__title"]');
    expect(itemSearch.length).toBe(1);
  });

  it('render search__items__found__infinite__scroll when search isn`t empty', () => {
    const component = setup({ search: 'fooBar' });
    const itemSearch = component.find('[data-test="search__items__found__infinite__scroll"]');
    expect(itemSearch.length).toBe(1);
  });

  it('not render search__items__found__infinite__scroll when search is empty', () => {
    const component = setup({ search: '' });
    const itemSearch = component.find('[data-test="search__items__found__infinite__scroll"]');
    expect(itemSearch.length).toBe(0);
  });

  it('render item__select-list', () => {
    const component = setup();
    const itemSearch = component.find('[data-test="item__select-list"]');
    expect(itemSearch.length).toBe(1);
  });

  it('render search__items__component not excluding wishlist items (wishlist empty)', () => {
    const component = setup({
      search: 'fooBar',
      data: {
        items: [
          { id: '1' },
          { id: '2' },
          { id: '3' },
          { id: '4' },
        ],
      },
      wishList: {
        items: [
        ],
      },
    });
    const itemSearch = component.find('[data-test="search__items__component"]');
    expect(itemSearch.props().items)
      .toEqual([
        { id: '1' },
        { id: '2' },
        { id: '3' },
        { id: '4' },
      ]);
  });

  it('render search__items__component excluding wishlist items', async () => {
    const storeRedux = storeFactory(initialState({
      search: 'fooBar',
      data: {
        items: [
          { id: '1' },
          { id: '2' },
          { id: '3' },
          { id: '4' },
        ],
      },
      wishList: {
        items: [
          { id: '1' },
          { id: '3' },
        ],
      },
    }));

    const component = shallow(
      <Search store={storeRedux} />,
    )
      .dive()
      .dive();

    const itemSearch = component.find('[data-test="search__items__component"]');
    expect(itemSearch.props().items)
      .toEqual([
        { id: '2' },
        { id: '4' },
      ]);
  });

  it('Scroll Infinite Scroll and call to getMorePages', async () => {
    const storeRedux = storeFactory(initialState({ search: 'fooBar' }));
    storeRedux.dispatch = jest.fn();
    const component = shallow(
      <Search store={storeRedux} />,
    )
      .dive()
      .dive();

    const itemSearch = component.find('InfiniteScroll');

    itemSearch.dive().instance().props.loadMore();

    expect(itemSearch.length).toBe(1);
    expect(storeRedux.dispatch).toHaveBeenCalledTimes(1);
    expect(storeRedux.dispatch.mock.calls[0][0].toString()).toBe(
      getMorePages().toString(),
    );
  });

  it('If not Scroll Infinite Scroll, no calls on getMorePages', async () => {
    const storeRedux = storeFactory(initialState({ search: 'fooBar' }));
    storeRedux.dispatch = jest.fn();
    const component = shallow(
      <Search store={storeRedux} />,
    )
      .dive()
      .dive();

    const itemSearch = component.find('InfiniteScroll');

    expect(itemSearch.length).toBe(1);
    expect(storeRedux.dispatch).toHaveBeenCalledTimes(0);
    expect(storeRedux.dispatch).not.toHaveBeenCalledWith(
      getMorePages,
    );
  });

  it('Changes redux items state when Scrolling Infinite Scroll', async () => {
    const storeRedux = storeFactory(initialState({ search: 'fooBar' }));
    const component = shallow(
      <Search store={storeRedux} />,
    )
      .dive()
      .dive();

    const mock = new MockAdapter(axios);
    const response = {
      items: [
        { id: 1 },
        { id: 2 },
        { id: 3 },
      ],
    };

    mock
      .onGet('https://www.googleapis.com/books/v1/volumes?q=fooBar&startIndex=0')
      .reply(200, response);

    const itemSearch = component.find('InfiniteScroll');
    await itemSearch.dive().instance().props.loadMore();
    await component.update();
    expect(storeRedux.getState().search).toBe('fooBar');
    expect(storeRedux.getState().data).toEqual(response);
  });
});
