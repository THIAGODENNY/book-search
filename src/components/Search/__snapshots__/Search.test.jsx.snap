// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render correctly  render search component correctly 1`] = `
<div
  className="search"
  data-test="search"
>
  <div
    className="search__search--found"
  >
    <input
      className="search__search__input-search"
      data-search="search__search__input-search"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      type="text"
      value="fooBar"
    />
  </div>
  <div
    className="search__items"
    data-test="search__items"
  >
    <div
      className="search__items__found"
      data-test="search__items__found"
    >
      <div>
        <div
          className="search__items__found__infinite__scroll"
          data-test="search__items__found__infinite__scroll"
        >
          <div
            className="component-items"
          />
        </div>
      </div>
    </div>
  </div>
  <div />
</div>
`;
